R/backend-oracle.R

```diff
@@ -45,9 +45,26 @@ sql_translate_env.Oracle <- function(con) {
       as.integer64  = sql_cast("NUMBER(19)"),
       as.numeric    = sql_cast("NUMBER"),
       as.double     = sql_cast("NUMBER"),
+      #as.Date = sql_cast("DATE"),
+      # Oracle date casting is NLS dependant, the following will work for the default YYYY-MM-DD
+      as.Date = function(x) sql_expr("DATE " !!x),
+
       # https://docs.oracle.com/cd/B19306_01/server.102/b14200/operators003.htm#i997789
       paste = sql_paste_infix(" ", "||", function(x) sql_expr(cast(!!x %as% text))),
       paste0 = sql_paste_infix("", "||", function(x) sql_expr(cast(!!x %as% text))),
+
+      # lubridate ---------------------------------------------------------------
+      # https://en.wikibooks.org/wiki/SQL_Dialects_Reference/Functions_and_expressions/Date_and_time_functions
+
+      today = function() sql_expr(TRUNC(SYSDATE)),
+      now = function() sql_expr(CURRENT_TIMESTAMP),
+
+      # https://modern-sql.com/feature/extract
+      yday = function(x) sql_expr((TO_NUMBER(TO_CHAR(!!x, "DDD")))),
+      wday = function(x) build_sql("(MOD(1 + TRUNC(", !!x, ") - TRUNC(", !!x, ", 'IW'),7) +1 )"),
```

Comment:
If you're using `build_sql()` you don't need `!!`

Reply:
Wasn't aware thanks, corrected in next commit